src/ripple/app/ledger/LedgerTiming.h

```diff
@@ -20,42 +20,108 @@
+    @param proposersValidated proposers who have validated the last closed ledger
+    @param previousMSeconds time, in milliseconds, for the previous ledger to
+                            reach consensus (in milliseconds)
+    @param currentMSeconds time, in milliseconds since the previous ledger closed
+    @param openMSeconds time, in milliseconds, since the previous LCL was computed
+    @param idleInterval the network's desired idle interval
+*/
+bool shouldCloseLedger (
+    bool anyTransactions,
+    int previousProposers,      int proposersClosed,    int proposerersValidated,
+    int previousMSeconds,       int currentMSeconds,    int openMSeconds,
+    int idleInterval);
+
+/** Determine whether the network reached consensus and whether we joined.
+
+    @param previousProposers proposers in the last closing (not including us)
+    @param currentProposers proposers in this closing so far (not including us)
+    @param currentAgree proposers who agree with us
+    @param currentFinished proposers who have validated a ledger after this one
+    @param previousAgreeTime how long it took to agree on the last ledger
+    @param currentAgreeTime how long we've been trying to agree
+
+    @return a pair of booleans:
+        * If the first boolean is false we don't have consensus and the second
+          value is ignored.
+        * If the first boolean is true then we have consensus and the second
+          boolen is true if we joined in the consesus, or false otherwise.
+*/
+std::pair<bool, bool> checkConsensus (
+    int previousProposers,      int currentProposers,
+    int currentAgree,           int currentClosed,
+    int previousAgreeTime,      int currentAgreeTime);
+
+/** Calculates the close time resolution for the specified ledger.
+
+    The Ripple protocol uses binning to represent time intervals using only one
+    timestamp. This allows servers to derive a common time for the next ledger,
+    without the need for perfectly synchronized clocks.
+    The time resolution (i.e. the size of the intervals) is adjusted dynamically
+    based on what happened in the last ledger, to try to avoid disagreements.
+*/
+int getNextLedgerTimeResolution (
+    int previousResolution,
+    bool previousAgree,
+    std::uint32_t ledgerSeq);
+
+//------------------------------------------------------------------------------
+
+// These are protocol parameters used to control the behavior of the system and
+// they should not be changed arbitrarily.
+
+// The percentage threshold above which we can declare consensus.
+int const minimumConsensusPercentage = 80;
+
+// Initial resolution of ledger close time.
+const int ledgerDefaultTimeResolution = 30;
+
+// All possible close time resolutions. This list MUST include the value
+// listed in ledgerDefaultTimeResolution and the first and last values must
+// be duplicated.
+int const ledgerPossibleTimeResolutions[] = { 10, 10, 20, 30, 60, 90, 120, 120 };
```

Comment:
Agreed about `chrono`.  I'm waiting for VS-2015 before making a significant `chrono` push because I want to do it right (with `constexpr`).